multiversx_sdk/entrypoints/entrypoints_test.py

```diff
@@ -173,3 +176,59 @@ def test_create_transfer_transaction_with_custom_gas_limit_and_gas_price(self):
 
         assert transaction.gas_limit == 10_000_000
         assert transaction.gas_price == 10_000_000_000_000
+
+    def test_initialize_entrypoint(self):
+        entrypoint = NetworkEntrypoint(
+            network_provider_url="https://devnet-api.multiversx.com",
+            network_provider_kind="api",
+            chain_id="D",
+        )
+        assert entrypoint.chain_id == "D"
+        assert isinstance(entrypoint.network_provider, ApiNetworkProvider)
+        assert entrypoint.network_provider.url == "https://devnet-api.multiversx.com"
+
+        entrypoint = NetworkEntrypoint(
+            network_provider_url="https://devnet-gateway.multiversx.com",
+            network_provider_kind="proxy",
+            chain_id="D",
+        )
+        assert entrypoint.chain_id == "D"
+        assert isinstance(entrypoint.network_provider, ProxyNetworkProvider)
+        assert entrypoint.network_provider.url == "https://devnet-gateway.multiversx.com"
+
+        with pytest.raises(InvalidNetworkProviderKindError):
+            entrypoint = NetworkEntrypoint(
+                network_provider_url="https://devnet-gateway.multiversx.com",
+                network_provider_kind="test",
+            )
+
+        api = ApiNetworkProvider("https://devnet-api.multiversx.com")
+        entrypoint = NetworkEntrypoint(network_provider=api)
+        assert entrypoint.chain_id is None
+        assert isinstance(entrypoint.network_provider, ApiNetworkProvider)
+        assert entrypoint.network_provider.url == "https://devnet-api.multiversx.com"
+
+        api = ApiNetworkProvider("https://devnet-api.multiversx.com")
+        entrypoint = NetworkEntrypoint(network_provider=api, chain_id="D")
+        assert entrypoint.chain_id == "D"
+        assert isinstance(entrypoint.network_provider, ApiNetworkProvider)
+        assert entrypoint.network_provider.url == "https://devnet-api.multiversx.com"
+
+        api = ApiNetworkProvider("https://devnet-api.multiversx.com")
+        entrypoint = NetworkEntrypoint.new_from_network_provider(api)
+        assert entrypoint.chain_id == "D"
+        assert isinstance(entrypoint.network_provider, ApiNetworkProvider)
+        assert entrypoint.network_provider.url == "https://devnet-api.multiversx.com"
+
+        api = ApiNetworkProvider("https://devnet-api.multiversx.com")
+        entrypoint = NetworkEntrypoint.new_from_network_provider(network_provider=api, chain_id="D")
+        assert entrypoint.chain_id == "D"
+        assert isinstance(entrypoint.network_provider, ApiNetworkProvider)
+        assert entrypoint.network_provider.url == "https://devnet-api.multiversx.com"
+
+    def test_ensure_chain_id_is_correctly_fetched(self):
+        api = ApiNetworkProvider("https://devnet-api.multiversx.com")
+        entrypoint = NetworkEntrypoint.new_from_network_provider(api)
+
+        delegation_controller = entrypoint.create_delegation_controller()
+        assert delegation_controller.factory.config.chain_id == "D"
```

Comment:
Why not `assert entrypoint.chain_id == "D"`?

Reply:
changed